src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -358,6 +363,68 @@ entails SolverOptions{..} constraint context hints =
         subclassDictionaryValue dict className index =
           App (Accessor (mkString (superclassName className index)) dict) valUndefined
 
+    solveCoercible :: Environment -> [Type] -> Maybe [TypeClassDict]
+    solveCoercible env [a, b] = do
+      let tySynMap = typeSynonyms env
+          replaceTySyns = either (const Nothing) Just . replaceAllTypeSynonymsM tySynMap
+      a' <- replaceTySyns a
+      b' <- replaceTySyns b
+      -- Solving terminates when the two arguments are the same. Since we
+      -- currently don't support higher-rank arguments in instance heads, term
+      -- equality is a sufficient notion of "the same".
+      if a' == b'
+        then pure [TypeClassDictionaryInScope [] 0 EmptyClassInstance [] C.Coercible [a, b] Nothing]
+        else do
+          -- When solving must reduce and recurse, it doesn't matter whether we
+          -- reduce the first or second argument -- if the constraint is
+          -- solvable, either path will yield the same outcome. Consequently we
+          -- just try the first argument first and the second argument second.
+          ws <- coercibleWanteds env a b <|> coercibleWanteds env b a
```

Comment:
Shouldn't these be `a'` and `b'`?

Reply:
I think you are probably right. I've fixed this.